include/deal.II/base/memory_space_data.h

```diff
@@ -25,47 +25,83 @@
 #include <functional>
 #include <memory>
 
+#ifdef DEAL_II_USE_KOKKOS_BACKEND
+#  include <Kokkos_Core.hpp>
+#endif
+
 DEAL_II_NAMESPACE_OPEN
 
 /**
  */
 namespace MemorySpace
 {
   /**
-   * Data structure
+   * Structure which stores data on the host or the device depending of the
+   * template parameter @p MemorySpace.
    */
   template <typename Number, typename MemorySpace>
   struct MemorySpaceData
   {
-    MemorySpaceData()
-    {
-      static_assert(std::is_same<MemorySpace, Host>::value ||
-                      std::is_same<MemorySpace, CUDA>::value,
-                    "MemorySpace should be Host or CUDA");
-    }
+    MemorySpaceData();
+
+    /**
+     * Return true if there is data stored on the host
+     */
+    bool
+    has_data();
```

Comment:
Is it possible for an object to only store data on the device? If so, is it possible that `has_data()==false`, but the object still stores data somewhere? If that were the case, I'd prefer renaming this function to `has_data_on_host()`.